Repeat Group UI Improvement

Jtang-1 opened the pull request:
## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->
Platform change for web apps:
This change will update the UI for repeat group with the main change being:
1. Having each repeat group item be collapsible with a colored header
2. Border surrounding each repeat group item
3. Replaced "X" icon with trash can icon so it does not get confused with "collapsing"
4. If a group uses a `group-collapse` appearance attribute but does not define a "Display text", it will no longer contain a border

With the following app config:
![Screen Shot 2024-08-01 at 12 48 54 PM](https://github.com/user-attachments/assets/1a724719-a116-4b4b-8aed-536fc9c77195)


Before
![Screen Shot 2024-08-01 at 12 48 07 PM](https://github.com/user-attachments/assets/1bb09476-ace2-42d2-8230-1377d9646f37)


After
![Screen Shot 2024-08-05 at 2 27 16 PM](https://github.com/user-attachments/assets/642683be-bec7-42b2-9c02-f0e53cc80e0c)

Change no. 4:
Before
![Screen Shot 2024-08-09 at 4 24 38 PM](https://github.com/user-attachments/assets/f68ff86f-b039-4b2c-87b4-8908d48af731)

After
![Screen Shot 2024-08-09 at 4 24 12 PM](https://github.com/user-attachments/assets/b76ddb25-8ec6-4516-930a-79ba5d36b72f)

## Technical Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
[Jira Ticket
](https://dimagi.atlassian.net/browse/USH-4383)
[Design Doc
](https://docs.google.com/document/d/1KMN6rCT9uZXdn4cRTlrIqPZTThupvYr47Kc0MdFLDEY/edit#heading=h.ipgou4bu0qoj)
When a repeat group is deleted, all its siblings are also removed from the JSON and the Knockout (KO) view model is rebuilt. Then, the siblings are added back, and the KO view model is rebuilt again. This process is done to force a refresh of the repeat groups. However, this causes the collapse state of the repeat group to revert to its default open state whenever one of its siblings is deleted.

To address this issue, we needed a way to retain the "collapse" state of the repeat groups. This PR introduces a solution where the Ix keys of the collapsed repeat groups are stored in session storage. During the forced refresh, the view model will check the collapsedIx session storage key and collapse the repeat groups if their Ix keys are present.

I also considered passing this state from Formplayer, but it seemed overly complicated and beyond FP's scope to track this UI state.

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->
No feature flag

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
locally tested and will test on staging. Change is limited to UI and most of the major changes are already existing styles for collapsed group

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->
no automated test

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
no QA

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

Jtang-1 committed:
feat: make repeat group items collapsible and default to open

Jtang-1 committed:
feat: make btn background white but stay red when hovered

Jtang-1 committed:
refactor: simplify logic

Jtang-1 committed:
rename variable

Jtang-1 committed:
feat: on repeat group deletion, all repeat groups are deleted and reapplied to force refresh.
This change preserves the collapse state of each repeat group on refresh

Jtang-1 committed:
clear stored collapsedIx when leaving form so it doesn't apply style when navigating to another form

Jtang-1 committed:
bug: remove nested repeat groups from session if the parent one if deleted

Jtang-1 committed:
sync scss with less

Jtang-1 committed:
"Bootstrap 5 Migration - Rebuilt diffs"

Jtang-1 committed:
Merge branch 'master' into jt/repeat-group-ui

MartinRiese commented on corehq/apps/hqwebapp/static/cloudcare/scss/formplayer-webapp/form.scss:
```diff
@@ -131,6 +131,15 @@ $group-indent: 15px;
     }
+
+    .btn {
+      background-color: white;
```
Does it just not look good with the background being red?

Jtang-1 replied:
Before this PR, it appears as white because the button's background is transparent. With the underlying background changing to blue, the button also becomes blue if i don't make it white.

Jtang-1 committed:
change icon from "X" to trash can for UX so it is not confused with "Collapsing"

Jtang-1 committed:
Merge branch 'master' into jt/repeat-group-ui

Jtang-1 committed:
bug: self.repeatable is not always an observable and do not overwrite instance property

Jtang-1 committed:
Merge branch 'master' into jt/repeat-group-ui

Jtang-1 committed:
fixes bug where repeat group or group (with collapsible display condition) do not have a display text,
the group header is hidden but the group still displays a border. Without the header shown, there is no
practical reason to still display as collapsible. In particular, there are use cases of repeat group
we do not want a header to be shown and it was still resuling in border surrounding each repeat group

Jtang-1 committed:
Merge branch 'master' into jt/repeat-group-ui

Jtang-1 committed:
"Bootstrap 5 Migration - Rebuilt diffs"
